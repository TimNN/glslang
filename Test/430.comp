#version 430 core

layout(local_size_x = 2) in;
layout(local_size_x = 16) in;     // ERROR, changing
layout(local_size_z = 4096) in;   // ERROR, too large
layout(local_size_x = 2) in;

const int total = gl_MaxComputeWorkGroupCount.y 
                + gl_MaxComputeUniformComponents
                + gl_MaxComputeTextureImageUnits
                + gl_MaxComputeImageUniforms
                + gl_MaxComputeAtomicCounters
                + gl_MaxComputeAtomicCounterBuffers;

void main()
{
    barrier();
    memoryBarrier();
    memoryBarrierAtomicCounter();
    memoryBarrierBuffer();
    memoryBarrierShared();
    memoryBarrierImage();
    groupMemoryBarrier();
}

layout(location = 2) in vec3 v3;      // ERROR
in float f;                           // ERROR
out float fo;                         // ERROR

shared vec4 s;
layout(location = 2) shared vec4 sl;  // ERROR
shared float fs = 4.2;                // ERROR

layout(local_size_x = 2, local_size_y = 3, local_size_z = 4) out;  // ERROR

int arrX[gl_WorkGroupSize.x];
int arrY[gl_WorkGroupSize.y];
int arrZ[gl_WorkGroupSize.z];
