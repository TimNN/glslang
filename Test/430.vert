#version 430 core

layout(location = 3) vec4 v4;  // ERROR

layout(location = 3) uniform vec4 uv4;

layout(location = 2) in   inb1 { vec4 v; } b1;  // ERROR
layout(location = 2) out outb1 { vec4 v; } b2;  // ERROR

out gl_PerVertex {
    float gl_ClipDistance[];
};

void foo()
{
    gl_ClipDistance[2] = 3.7;
}
