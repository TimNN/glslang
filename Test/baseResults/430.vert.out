Warning, version 430 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:3: 'v4' : location qualifiers only appy to uniform, buffer, in, or out storage qualifiers 
ERROR: 0:7: 'location qualifier on input block' : not supported for this version or the enabled extensions 
ERROR: 0:8: 'location qualifier on output block' : not supported for this version or the enabled extensions 
ERROR: 3 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:?   Linker Objects
0:?     'v4' (layout(location=3 ) 4-component vector of float)
0:?     'uv4' (layout(location=3 ) uniform 4-component vector of float)
0:?     'b1' (layout(location=2 ) in block)
0:?     'b2' (layout(location=2 ) out block)
0:?     'gl_VertexID' (gl_VertexId int)
0:?     'gl_InstanceID' (gl_InstanceId int)

